spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -104,7 +104,7 @@ public DataSourceReader createReader(StructType readSchema, DataSourceOptions op
     validateWriteSchema(table.schema(), dsSchema, checkNullability(options));
     validatePartitionTransforms(table.spec());
     String appId = lazySparkSession().sparkContext().applicationId();
-    String wapId = lazySparkSession().conf().get("spark.wap.id", null);
+    String wapId = options.get("write-wap-id").orElse(lazySparkSession().conf().get("spark.wap.id", null));
```

Comment:
How about "wap-id"? The "write" context doesn't add much clarity.

Reply:
Can we move this change to a separate commit? I don't think it needs to be part of cherry-picking.

Reply:
will do